conans/model/layout.py

```diff
@@ -27,16 +26,22 @@ def __init__(self):
 
 class Infos(object):
 
-    def __init__(self):
-
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
         self.source = NewCppInfo()
         self.build = NewCppInfo()
         self.package = NewCppInfo()
 
+    @property
+    def build_info(self):
+        build_cppinfo = self._conanfile.cpp.build.copy()
+        build_cppinfo.set_relative_base_folder(self._conanfile.folders.build_folder)
```

Comment:
I didn't reuse this in the ``installer.py``, because there it calls:

```python
build_cppinfo.set_relative_base_folder(self._conanfile.folders.build)
```
Instead of ``conanfile.folders.build_folder``, so there it is still a relative path, while here we need an absolute path.